### PR TITLE
set time bounds

### LINKED ISSUE #176: create time bounds after time averaging
Missing time bounds (`time_bnds`) variable after data is resampled to new frequency (i.e., time averaging step).

After introducing this feature, the netcdf file should roughly look like this

```
netcdf file_with_time_bounds {
dimensions:
	time = UNLIMITED ; // (2 currently)
	lat = 3 ;
	lon = 4 ;
	bnds = 2 ;   # <-- this entry
variables:
	double time(time) ;
		time:bounds = "time_bnds" ;      # <-- this entry
		time:units = "days since 2018" ;
		time:calendar = "360_day" ;
		time:axis = "T" ;
		time:long_name = "time" ;
		time:standard_name = "time" ;
	double time_bnds(time, bnds) ;      # <-- this entry
```

### PR DESCRIPTION
**Pull request by siligam**
# Time Bounds Feature Implementation

## Overview
This PR introduces a robust time bounds handling system for xarray Datasets, which is particularly useful for climate and weather data where variables are often associated with time intervals rather than single time points.

## Key Features
- 🚀 Automatic detection and handling of time coordinates
- 📅 Support for various time frequencies (daily, monthly, custom offsets)
- ✅ Comprehensive input validation
- 📊 Detailed logging with markdown-style formatting
- 🧪 Extensive test coverage including edge cases

## Changes
1. **Core Functionality**:
   - Implemented [time_bounds](cci:1://file:///Users/pasili001/repos/pymorize/src/pymor/std_lib/time_bounds.py:7:0-59:62) function in [pymor/std_lib/time_bounds.py](cci:7://file:///Users/pasili001/repos/pymorize/src/pymor/std_lib/time_bounds.py:0:0-0:0)
   - Added support for both standard and offset time points
   - Included automatic bounds creation with proper dimension handling

2. **Testing**:
   - Added unit tests for various scenarios
   - Included tests for monthly frequency with and without offsets
   - Added validation for error cases

3. **Documentation**:
   - Created comprehensive documentation in [doc/time_bounds.rst](cci:7://file:///Users/pasili001/repos/pymorize/doc/time_bounds.rst:0:0-0:0)
   - Added usage examples and API reference
   - Documented logging format and error handling

## Testing
All tests are passing:
- Tested with different time frequencies
- Verified edge cases (single time point, existing bounds)
- Confirmed proper error handling

## Logging 
```
[Time bounds] dataset_name
  → is dataset:   ✅
  → time label  : time
  → bounds label: bnds
  → has time bounds: ❌
  → time values: 365 points from 2000-01-01 to 2000-12-31
  → time step: 1 days 00:00:00
  → set time bounds: time_bnds(365, 2)
  → bounds range: 2000-01-01 to 2001-01-01
```

## Next Steps
- [ ] Review implementation
- [ ] Refine logic by including infer_frequency from PR #186 
- [ ] Update any dependent documentation

## Related Issues
Closes #176

### CODE REVIEW

**Comment by pgierz** [comment]
@siligam this is still marked as a draft, is there anything to be done here, or can I review this and merge? Is it still required for prep release?

**Comment by siligam** [comment]
> @siligam this is still marked as a draft, is there anything to be done here, or can I review this and merge? Is it still required for prep release?

it is ready for review, although the linting thing needs to be fixed.

File: src/pymor/std_lib/time_bounds.md
Comment (pgierz) [inline src/pymor/std_lib/time_bounds.md:1]: this file belongs in the documentation folder, together with the time_bounds.rst and should be merged into one single document

File: src/pymor/std_lib/time_bounds.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    rule : Rule
+        Rule object containing time bounds file attribute and approx_interval
+
+    Returns
+    -------
+    xr.Dataset
+        The output dataset with the time bounds information.
+    """
+    # Get dataset name for logging
+    dataset_name = ds.attrs.get("name", "unnamed_dataset")
```
Comment (pgierz) [inline src/pymor/std_lib/time_bounds.py:31]: We have this in a couple of places. I think we should be consistent and think of a rule for it. I am ok with `"unnamed_dataset"`, but we should keep it in mind for the future.

File: src/pymor/std_lib/time_bounds.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+    # Log header with markdown style
+    logger.info(f"[Time bounds] {dataset_name}")
+    logger.info(f"  → is dataset:   {'✅' if isinstance(ds, xr.Dataset) else '❌'}")
```
Comment (pgierz) [inline src/pymor/std_lib/time_bounds.py:35]: We have this in a few places now, and I am not sure I like the emojis so much. What about @mandresm, any opinions?

File: src/pymor/std_lib/time_bounds.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    if time_method not in ["mean", "instantaneous", "climatology"]:
+        logger.warning(
+            f"  ⚠️  Unknown time method '{time_method}', defaulting to 'mean'"
+        )
+        time_method = "mean"
+
+    # Only create bounds for mean and instantaneous methods
+    if time_method == "climatology":
+        logger.info("  → skipping bounds creation for climatology data")
+        return ds
```
Comment (pgierz) [inline src/pymor/std_lib/time_bounds.py:79]: Can we double check this? Do climatologically averaged datasets really not include time bounds? That seems inconsistent.

File: src/pymor/std_lib/time_bounds.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    if has_time_bounds:
+        logger.info(f"  → using existing bounds: {time_bounds_label}")
+        return ds
+
+    # Validate time method
+    if time_method not in ["mean", "instantaneous", "climatology"]:
+        logger.warning(
+            f"  ⚠️  Unknown time method '{time_method}', defaulting to 'mean'"
+        )
+        time_method = "mean"
```
Comment (pgierz) [inline src/pymor/std_lib/time_bounds.py:74]: Having a way to bail out here and raise a failure would be nice.

File: src/pymor/std_lib/time_bounds.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    # Log time values info
+    logger.info(
+        f"  → time values  : {len(time_values)} points from {time_values[0]} to {time_values[-1]}"
+    )
+
+    # Handle instantaneous time method
+    if time_method == "instantaneous":
+        logger.info("  → creating instantaneous bounds (delta time = 0)")
+        # For instantaneous data, bounds are the same as time values
+        bounds_data = np.column_stack([time_values, time_values])
```
Comment (pgierz) [inline src/pymor/std_lib/time_bounds.py:98]: As a reminder of what `np.column_stack` does:
```suggestion
        # For instantaneous data, bounds are the same as time values
        #
        # .. note:: np.column_stack reminder
        #
        #    >>> a = np.array([1, 2, 3])
        #    >>> b = np.array([4, 5, 6])
        #    >>> np.column_stack((a, b))
        #    [[1, 4], [2, 5], [6, 3]]
        #
        bounds_data = np.column_stack([time_values, time_values])
```

File: src/pymor/std_lib/time_bounds.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    else:  # time_method == 'mean'
+        logger.info("  → creating mean bounds (delta time > 0)")
+
+        if len(time_values) < 2:
+            error_msg = "Cannot create mean time bounds: need at least 2 time points"
+            logger.error(f"  ❌ {error_msg}")
+            raise ValueError(error_msg)
+
+        # Calculate data frequency in days
+        time_diff_seconds = np.median(
```
Comment (pgierz) [inline src/pymor/std_lib/time_bounds.py]: Why median?

File: src/pymor/std_lib/time_bounds.py [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+            )
+
+    # Create the bounds DataArray
+    bounds = xr.DataArray(
+        data=bounds_data,
+        dims=(time_label, bounds_dim_label),
+        coords={time_label: time_values, bounds_dim_label: [0, 1]},
+        attrs={
+            "long_name": f"time bounds for {time_label}",
+            "comment": f"Generated by pymorize: {time_method} time bounds",
```
Comment (pgierz) [inline src/pymor/std_lib/time_bounds.py:144]: wrong name!

**Comment by pgierz** [comment]
Superseded by #264, which reimplements this on the current `pycmor` namespace (the original targeted the old `pymor` naming). Logic is preserved, tests expanded.
